CMakeLists.txt

```diff
@@ -261,6 +261,10 @@ if( UNIX )
         set(CPACK_DEBIAN_PACKAGE_ARCH "${DEBIAN_PACKAGE_ARCH}")
         set(CPACK_DEBIAN_PACKAGE_PRIORITY "optional")
 
+        set(CPACK_DEBIAN_PACKAGE_CONTROL_EXTRA "${CMAKE_CURRENT_SOURCE_DIR}/debian/postinst;${CMAKE_CURRENT_SOURCE_DIR}/debian/postrm")
+
+        set(CPACK_DEBIAN_PACKAGE_DEPENDS "amcl(>= 4.1.1), xaptum-tpm(>= 0.4.0), libsodium18(>= 1.0.8)")
```

Comment:
Use `set(CPACK_DEBIAN_PACKAGE_SHLIBDEPS ON)` to automatically populate the `..._PACKAGE_DEPENDS` variable.

https://cmake.org/cmake/help/v3.3/module/CPackDeb.html#variable:CPACK_DEBIAN_PACKAGE_SHLIBDEPS

Reply:
I can't seem to get the `SHLIBDEPS` version to find the `amcl` dependency, even though `ldd` shows the library dependence and `dpkg -S` correctly gives the `amcl` deb as the source of the amcl libraries.

AMCL is also a bit of an oddball dependency, because it's not just the shared library that we depend on, it's also the headers.

So, I think I'm going to add the `SHLIBDEPS`, but keep the explicit `amcl` dependence (the `xaptum-tpm` dependence actually shouldn't be there, because that *should* get statically linked in).

Reply:
FWIW, I realized the `amcl` dependency only existed for the `-dev` version (the usual version, with just the shared lib, just has the AMCL libs statically-linked in and we don't expect anyone to need the AMCL headers, since it's not a `-dev` package).

So, only the `-dev` package has the `amcl` dependence.